utils/build.ps1

```diff
@@ -583,21 +583,39 @@ function Invoke-BuildStep {
     [Parameter(Position=1, Mandatory)]
     [Hashtable] $Platform,
     [Parameter(ValueFromRemainingArguments)]
-    $RemainingArgs
+    [Object[]] $RemainingArgs
   )
 
   if ($Summary) {
     $Stopwatch = [Diagnostics.Stopwatch]::StartNew()
   }
 
   $SplatArgs = @{}
-  foreach ($Arg in $RemainingArgs) {
-    if ($Arg -is [Hashtable]) {
-      $SplatArgs += $Arg
-    } elseif ($Arg -is [string]) {
-      $SplatArgs[$Arg.TrimStart('-')] = $true
-    } else {
-      throw "$Arg is unknown type: $($Arg.GetType())"
+  if ($RemainingArgs) {
+    $i = 0
+    while ($i -lt $RemainingArgs.Count) {
```

Comment:
Why not use an iterator?

Reply:
`$RemainingArgs.GetEnumerator()` would get you get the arguments themselves without having to index.